web-components/src/mixins/FocusTrapMixin.ts

```diff
@@ -361,33 +375,51 @@ export const FocusTrapMixin = <T extends AnyConstructor<FocusClass & FocusTrapCl
         this.manageNewElement(focusableElement as HTMLElement);
       }
     }
+
     updateFocusableElements = () => {
-      if(this.focusableTimer) {
-        clearTimeout(this.focusableTimer)
-        this.focusableElements = []
+      if (this.focusableTimer) {
+        clearTimeout(this.focusableTimer);
+        this.focusableElements = [];
       }
       this.focusableTimer = setTimeout(() => {
         this.setFocusableElements();
       }, 10);
-    }
+    };
+
+    handleChildFocusTrap = (event: CustomEvent) => {
+      if (event.target !== this) {
```

Comment:
Why this is required. Please explain the use case.

Reply:
The use case is, when we try to show a overlay on top of another where the focus is trapped ( e.g. Calendar control over a Modal ) the focus is always trapped on to the modal and not the uppermost calendar. With these changes, as soon as the calendar acquires focus trap, the modal will forgoes the focus trap based on the event emitted by child - calendar. As soon as child forgoes focus, the parent if any will re-acquire the focus.

Since this method is on generic mixin, now when the calendar emits focus-untrap event the same should not be consumed by itself as it also listens to the same event and assume some other child has forgone the focus and re-acquire itself. Hence this check to avoid such cases.

Reply:
Have we tested other use cases as well? If modal contains some buttons along with Menu Overlay, once menu overlay is opened, will the focus be trapped inside Menu Overlay only? As there can be a use case where we may need to trap focus only inside Modal

Reply:
Yes. These scenarios have been tested